### PR TITLE
C# in Tag Helpers: explain false attribute values

### LINKED ISSUE #24860: False-valued-attribute-erasing magic is undocumented
This code does not make any sense:
```cshtml
<input asp-for="LastName" 
       disabled="@(Model?.LicenseId == null)" />
```

It is functionally equivalent to the following code:
```cshtml
<input asp-for="LastName" disabled>
```

&hellip; unless you *know* that Razor removes the attribute if its *computed* value is `false`, which is undocumented.  This makes the attribute declarations `disabled=false` and `disabled="@false"` work differently, which is so unexpected that it begs an explanation.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 62cdbf88-1fff-6999-f75a-26fbe730b911
* Version Independent ID: bc48b5cb-d500-f75d-8e48-be4d069a9198
* Content: [Tag Helpers in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/intro?view=aspnetcore-6.0)
* Content Source: [aspnetcore/mvc/views/tag-helpers/intro.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/views/tag-helpers/intro.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by yecril71pl**
Fixes #24860

### CODE REVIEW

**Comment by guardrex** [comment]
@serpent5 ... Do you mind a quick 👁️ here?

I was going to ping Rick, but he's 😴🛌 at the moment ... in hammock on a beach ⛱️😄.

**Review by serpent5 — APPROVED** [review]

**Comment by serpent5** [comment]
The changes look good, and the @guardrex edits are superb. I'm not sure what's going on with the CLA service, but I'll see if I can find out.

**Comment by guardrex** [comment]
Yeah ... I just noticed that. That rarely breaks, but the *gremlins* 😈 may have gotten into it this morning!

**Comment by serpent5** [comment]
This CLA issue is widespread, affecting many teams. I'll keep 👁️ the internal discussion to see how it's progressing.

**Comment by guardrex** [comment]
Ok ... I assigned but perhaps it's better to leave it open for any of us to pull the trigger on.

File: aspnetcore/mvc/views/tag-helpers/intro.md [hunk 1]
```diff
@@ -153,6 +153,8 @@ The preceding code can be written as:
        disabled="@(Model?.LicenseId == null)" />
 ```
 
+Normally, the `@` operator inserts a textual representation of an expression into the rendered HTML markup. However, when an expression evaluates to logical `false`, the framework removes the attribute. In the preceding example, the `disabled` attribute is removed if `Model` or `LicenseId` is `null`.
```
Comment (yecril71pl) [inline aspnetcore/mvc/views/tag-helpers/intro.md:156]: How about "removes the attribute instead"?

[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/mvc/views/tag-helpers/intro.md:156]: Thanks @yecril71pl ... I'll take a look in a bit. I'm buried for the next few hours. ⛰️⛏️😨

[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/mvc/views/tag-helpers/intro.md]: Got it!

**Pull request merged**